python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,250 @@
+        storageClass="AstropyQTable",
+        name="zernikes",
+    )
+    donutQualityTable = connectionTypes.Output(
+        doc="Quality information for donuts",
+        dimensions=("visit", "detector", "instrument"),
+        storageClass="AstropyQTable",
+        name="donutQualityTable",
+    )
+
+
+class CalcZernikesNeuralTaskConfig(
+    pipeBase.PipelineTaskConfig,
+    pipelineConnections=CalcZernikesNeuralTaskConnections,  # type: ignore
+):
+    """Configuration for CalcZernikesNeuralTask.
+
+    Attributes
+    ----------
+    wavenet_path : str
+        Model Weights Path for wavenet
+    alignet_path : str
+        Model Weights Path for alignet
+    aggregatornet_path : str
+        Model Weights Path for aggregatornet
+    dataset_param_path : str
+        datasetparam path
+    device : str
+        Device to use for calculations
+    """
+
+    wavenet_path = pexConfig.Field(  # type: ignore
+        doc="Model Weights Path for wavenet",
+        dtype=str
+    )
+    alignet_path = pexConfig.Field(  # type: ignore
+        doc="Model Weights Path for alignet",
+        dtype=str
+    )
+    aggregatornet_path = pexConfig.Field(  # type: ignore
+        doc="Model Weights Path for aggregatornet",
+        dtype=str
+    )
+    dataset_param_path = pexConfig.Field(  # type: ignore
+        doc="datasetparam path",
+        dtype=str
+    )
+    device = pexConfig.Field(  # type: ignore
+        doc="Device to use for calculations",
+        dtype=str,
+        default="cuda"
+    )
+
+
+class CalcZernikesNeuralTask(pipeBase.PipelineTask):
+    """Neural network-based Zernike estimation task using TARTS.
+
+    This class uses the TARTS (Telescope Active Optics Real-Time System)
+    neural network models to estimate Zernike coefficients from pairs of
+    intra and extra-focal exposures.
+    """
+    _DefaultName = "calcZernikesNeuralTask"
+
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+
+        # Define default Noll indices (zk terms 4-23, excl piston, tip, tilt)
+        self.nollIndices = np.arange(4, 24)
+
+        self.tarts = NeuralActiveOpticsSys(
+            self.config.dataset_param_path,  # type: ignore
+            self.config.wavenet_path,  # type: ignore
+            self.config.alignet_path,  # type: ignore
+            self.config.aggregatornet_path  # type: ignore
+        )
+        self.tarts = self.tarts.eval()
+        self.CROP_SIZE = self.tarts.CROP_SIZE
+        self.device = self.config.device  # type: ignore
+
+        if self.device == "cpu":
+            self.tarts = self.tarts.cpu()
+
+            # Update device attributes for all sub-models to CPU
+            self.tarts.device_val = torch.device("cpu")
+            if hasattr(self.tarts.alignnet_model, 'device_val'):
+                self.tarts.alignnet_model.device_val = torch.device("cpu")
+            if hasattr(self.tarts.wavenet_model, 'device_val'):
```

Comment:
Just so I understand, is it possible that each model does not have device_val? If one has it, does that mean that all of them have it?